### PR TITLE
Feature/multiupload breadcrumb

### PR DESCRIPTION
**Pull request by boubaker**

### CODE REVIEW

File: integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java [hunk 1]
```diff
@@ -843,12 +863,52 @@ public DriveData getDocDrive(int i) {
...
+                  User user = organizationService.getUserHandler().findUserByName(userId);
+                  if (user != null) {
+                    driveName = user.getDisplayName();
+                  }
+                } catch (Exception e) {
+                  LOG.warn("Can't get drive name for user with id '" + userId + "'", e);
+                }
+              }
+            }
+          }
```
Comment (thomasdelhomenie) [inline integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java]: It seems we do not have unit tests for this class yet. It is probably time to add it in order to validate all this logic. WDYT ?

File: integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/UISharedFile.gtmpl [hunk 2]
```diff
@@ -301,6 +301,7 @@
 		def docPreviewUri = "javascript:require(['SHARED/documentPreview'], function(documentPreview) {documentPreview.init({doc: {id:'" + currentNode.getUUID() + "', repository:'" + repository + "', workspace:'" + preferenceWS + "', path:'" + currentNode.getPath() + "', title:'" + currentNode.getName() + "', downloadUrl:'" + docDownloadUrl + "', openUrl:'" + docOpenUri + "'}, author: {username:'" + ownerIdentity.getRemoteId() + "', fullname:'" + escapedOwnerName + "', avatarUrl:'" + ownerAvatar + "', profileUrl:'" + ownerUri + "'}, activity: {id: '" + activity.id + "', postTime:'" + escapedActivityPostedTime + "', status: '" + escapedActivityStatus + "', liked: " + uicomponent.isLiked() + ", likes: " + identityLikesNum + ", spaceURL: '" + spaceURL + "'}})})";
 		def docFolderRelativePath = uicomponent.getDocFolderRelativePath(0);
 		def docFolderBreadCrumb = uicomponent.getDocFolderRelativePathWithLinks(0);
+        def breadCrumbSize = docFolderBreadCrumb.size();
```
Comment (thomasdelhomenie) [inline integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/UISharedFile.gtmpl]: Should check for nullability of docFolderBreadCrumb.

File: integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl [hunk 3]
```diff
@@ -353,6 +353,34 @@ for(def index = 0; index < uicomponent.getFilesCount(); index ++) {
...
+            filePath = folderPath;
+        } else {
+	        if(folderIndex > 0) {
+	            breadCrumbContent += ",";
+	        }
+	        breadCrumbContent += "'" + folderName + "': '" + folderPath + "'";
+	        breadCrumbContent = breadCrumbContent.replace("%27", "\\'");
+	    }
+        folderIndex++;
+    }
```
Comment (thomasdelhomenie) [inline integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl]: We should move this piece of code in the java component.

File: integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl [hunk 4]
```diff
@@ -412,41 +439,100 @@ for(def index = 0; index < uicomponent.getFilesCount(); index ++) {
...
+                    def iconClass = "uiBgdFile";
+                    if (contentName.toLowerCase().contains(".pdf")) {
+                        iconClass = "uiBgdFilePDF";
+                    } else if (contentName.toLowerCase().contains(".doc")) {
+                        iconClass = "uiBgdFileWord";
+                    } else if (contentName.toLowerCase().contains(".xls")) {
+                        iconClass = "uiBgdFileExcel";
+                    } else if (contentName.toLowerCase().contains(".ppt")) {
+                        iconClass = "uiBgdFilePPT";
+                    }
```
Comment (thomasdelhomenie) [inline integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl]: We should move this piece of code in the java component.

**Comment by thomasdelhomenie** [comment]
I also noticed that the changes have not been implemented (at least not totally) for Share File activity type:
screenshot from 2017-11-16 14-08-55

File: integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java [hunk 5]
```diff
@@ -216,7 +222,11 @@ public String getContentName(int i) {
       return null;
     }
     ActivityFileAttachment activityFileAttachment = activityFileAttachments.get(i);
-    return activityFileAttachment.getContentName();
+    try {
+      return activityFileAttachment.getContentNode().getName();
```
[the remark this answers is not in the document]
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java]: Yes, I 've found a bug with ".getContentName()" which is retrieved from Activity parameters.
In fact when we rename the document, the name is not updated on AS, thus we should use the concrete name instead.

**Comment by boubaker** [comment]
> I also noticed that the changes have not been implemented (at least not totally) for Share File activity type:

Expected, cc @hghorbel

File: integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java]: ok. Shouldn't we change the content of the getContentName() to always return getContentNode().getName() instead ?

[the remark this answers is not in the document]
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java]: done,  getContentName() is used as fallback

[the remark this answers is not in the document]
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java]: done

File: integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java [hunk 6]
```diff
@@ -843,12 +863,52 @@ public DriveData getDocDrive(int i) {
...
         try {
+          Map<String, String> parameters = drive.getParameters();
+          String driveName = drive.getName();
+          if (parameters != null) {
+            if (parameters.containsKey("groupId")) {
+              String groupId = parameters.get("groupId");
+              if (StringUtils.isNotBlank(groupId)) {
+                try {
+                  if (groupId.startsWith(SpaceUtils.SPACE_GROUP)) {
+                    SpaceService spaceService = getApplicationComponent(SpaceService.class);
```
[the remark this answers is not in the document]
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java]: done

File: integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/UISharedFile.gtmpl (hunk 2 above)
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/UISharedFile.gtmpl]: done

File: integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl (hunk 3 above)
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl]: done

File: integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl (hunk 4 above)
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/resources/groovy/ecm/social-integration/plugin/space/FileUIActivity.gtmpl]: done

File: integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java (hunk 1 above)
Comment (boubaker) [reply integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/FileUIActivity.java]: done

**Comment by boubaker** [comment]
> @boubaker it seems the unit test for FileUIActivity has not been pushed

Oops, right ! done !

**Pull request merged**